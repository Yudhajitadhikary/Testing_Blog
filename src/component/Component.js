import React, { Component } from "react";
import ReactToPrint from "react-to-print";
import CardPrint from "../print/ComponentPrint";
import Component2 from "./Component2";
import { Formik, Field, Form, ErrorMessage } from "formik";
import * as Yup from "yup";
import { submitAction } from "../actions";
import { connect } from "react-redux";
class Card extends Component {
  constructor(props) {
    super(props);
    this.state = { value: "", count: 0, childCount: 0 };
    this.handleClick = this.handleClick.bind(this);
    this.childrenClick = this.childrenClick.bind(this);
  }
  handleClick() {
    this.setState({ count: this.state.count + 1 });
    document.getElementById("testId").style.display = "";
  }
  childrenClick() {
    this.setState({ childCount: this.state.childCount + 1 });
    document.getElementById("testId").style.display = "";
  }
  componentDidUpdate(PrevState) {
    console.log("Component Updated");
  }
  componentWillMount() {
    console.log("Component will mount");
  }
  componentDidMount() {
    console.log("Component did Mount");
  }

  render() {
    const {
      email,
      password,
      submitAction,
      phoneno,
      address,
      pincode,
    } = this.props;

    return (
      <div>
        <ReactToPrint
          trigger={() => (
            <button
              style={{
                textAlign: "center",
                marginLeft: "50%",
                backgroundColor: "lemonchiffon",
                width: "62px",
              }}
            >
              Print
            </button>
          )}
          content={() => this.componentRef}
        />
        <div style={{ display: "none" }}>
          <CardPrint
            email={email}
            password={password}
            phoneno={phoneno}
            address={address}
            pincode={pincode}
            ref={(el) => (this.componentRef = el)}
          />
        </div>
        <div>
          {" "}
          <div id="testId">{this.state.childCount}</div>
          <div id="testOnClick" onClick={this.handleClick}>
            Total Count
          </div>
        </div>
        <Component2 count={this.state.count} action={this.childrenClick} />
        <div>
          <Formik
            initialValues={{
              email: "",
              password: "",
              confirmPassword: "",
              pincode: "",
              address: "",
              phoneno: "",
            }}
            validationSchema={Yup.object().shape({
              email: Yup.string()
                .email("Email is invalid")
                .required("Email is required"),
              password: Yup.string()
                .min(6, "Password must be at least 6 characters")
                .matches(/[A-Z]{2}\d{2}[A-Z]{2}\d{4}$/i, "invalid Password")
                .required("Password is required"),
              confirmPassword: Yup.string()
                .oneOf([Yup.ref("password"), null], "Passwords must match")
                .required("Confirm Password is required"),
              pincode: Yup.string()
                // .max(6,'Pincode must be of 6 characters')
                .matches(/^[0-9]{6}$/, "Should be a number of length 6")
                .required("Pincode is required"),
              phoneno: Yup.string()
                // .max(6,'Pincode must be of 6 characters')
                .matches(/^[0-9]{10}$/, "Should be a number of length 10")
                .required("Contact Number is required"),
              address: Yup.string().required("Address is Required"),
            })}
            onSubmit={(fields) => {
              // console.log(fields)
              submitAction(fields);
            }}
            render={({ errors, status, touched }) => (
              <Form>
                <div className="form-group">
                  <label htmlFor="email">Email</label>
                  <Field
                    name="email"
                    type="text"
                    className={
                      "form-control" +
                      (errors.email && touched.email ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="email"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="password">Password</label>
                  <Field
                    name="password"
                    type="password"
                    className={
                      "form-control" +
                      (errors.password && touched.password ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="password"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="confirmPassword">Confirm Password</label>
                  <Field
                    name="confirmPassword"
                    type="password"
                    className={
                      "form-control" +
                      (errors.confirmPassword && touched.confirmPassword
                        ? " is-invalid"
                        : "")
                    }
                  />
                  <ErrorMessage
                    name="confirmPassword"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="address">Address</label>
                  <Field
                    name="address"
                    type="address"
                    className={
                      "form-control" +
                      (errors.address && touched.address ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="address"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="pincode">Pincode</label>
                  <Field
                    name="pincode"
                    type="pincode"
                    className={
                      "form-control" +
                      (errors.pincode && touched.pincode ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="pincode"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="phoneno">Contact Number</label>
                  <Field
                    name="phoneno"
                    type="phoneno"
                    className={
                      "form-control" +
                      (errors.phoneno && touched.phoneno ? " is-invalid" : "")
                    }
                  />
                  <ErrorMessage
                    name="phoneno"
                    component="div"
                    className="invalid-feedback"
                  />
                </div>
                <div className="form-group">
                  <button
                    type="submit"
                    style={{ marginLeft: "50%" }}
                    className="btn btn-primary mr-2"
                  >
                    Register
                  </button>
                </div>
              </Form>
            )}
          />
        </div>
      </div>
    );
  }
}
export const mapStateToProps = (state) => ({
  email: state.email,
  password: state.password,
  address: state.address,
  pincode: state.pincode,
  phoneno: state.phoneno,
});

export const mapDispatchToProps = (dispatch) => ({
  submitAction: (data) => dispatch(submitAction(data)),
});
export default connect(mapStateToProps, mapDispatchToProps)(Card);
