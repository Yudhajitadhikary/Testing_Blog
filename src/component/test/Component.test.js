import React from "react";
import { mount } from "../enzyme";
import { shallowToJson } from "enzyme-to-json";
import Component from '../Component';
import { mapStateToProps, mapDispatchToProps } from "../Component";
 import { Provider } from 'react-redux';
 import store from '../../store';
let wrapper;
it("should render TextFieldAtom component", () => {
  wrapper = mount(<Provider store={store}><Component/></Provider>, { attachTo: document.body });
  expect(shallowToJson(wrapper)).toMatchSnapshot();
  wrapper.find('#testOnClick').simulate('click')
  shallowToJson(wrapper).children[0].children[0].node.instance.handleClick()
  shallowToJson(wrapper).children[0].children[0].node.instance.childrenClick()
  shallowToJson(wrapper).children[0].children[0].children[0].children[0].props.content()
  shallowToJson(wrapper).children[0].children[0].children[0].children[4].children[0].props.onSubmit({email:'',password:'',address:'',pincode:''})
  wrapper.setProps({})
//   console.log(shallowToJson(wrapper).children[0])
//   console.log(shallowToJson(wrapper).node.instance.childrenClick)
  // shallowToJson(wrapper).node.instance.childrenClick()
});
it("should map state to props", () => {
  const initialState = {
    email: '',
    password:'',
    address:'',
    pincode:'',
    phoneno:''
  };
  expect(mapStateToProps(initialState).email).toEqual('');
});
it("should call mapDispatchToProps", () => {
  const dispatch = jest.fn();
  mapDispatchToProps(dispatch).submitAction();
});