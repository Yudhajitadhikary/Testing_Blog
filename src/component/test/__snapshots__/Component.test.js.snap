// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TextFieldAtom component 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Card)>
    <Card
      address=""
      email=""
      password=""
      phoneno=""
      pincode=""
      submitAction={[Function]}
    >
      <div>
        <d
          content={[Function]}
          copyStyles={true}
          pageStyle="@page { size: auto;  margin: 0mm; } @media print { body { -webkit-print-color-adjust: exact; } }"
          removeAfterPrint={false}
          suppressErrors={false}
          trigger={[Function]}
        >
          <button
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "lemonchiffon",
                "marginLeft": "50%",
                "textAlign": "center",
                "width": "62px",
              }
            }
          >
            Print
          </button>
        </d>
        <div
          style={
            Object {
              "display": "none",
            }
          }
        >
          <CardPrint
            address=""
            email=""
            password=""
            phoneno=""
            pincode=""
          >
            <div>
              <div
                style={
                  Object {
                    "color": "green",
                    "fontSize": "30px",
                    "paddingLeft": "30px",
                    "textAlign": "center",
                  }
                }
              >
                Here's a print copy of data you entered :)
              </div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "padding": "20px 20px 20px 50px ",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "color": "Red",
                      "fontSize": "22px",
                      "paddingRight": "10px",
                    }
                  }
                >
                  Email:
                </div>
                <div
                  style={
                    Object {
                      "fontSize": "22px",
                    }
                  }
                />
              </div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "padding": "20px 20px 20px 50px ",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "color": "Red",
                      "fontSize": "22px",
                      "paddingRight": "10px",
                    }
                  }
                >
                  Password:
                </div>
                <div
                  style={
                    Object {
                      "fontSize": "22px",
                    }
                  }
                />
              </div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "padding": "20px 20px 20px 50px ",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "color": "Red",
                      "fontSize": "22px",
                      "paddingRight": "10px",
                    }
                  }
                >
                  Pincode:
                </div>
                <div
                  style={
                    Object {
                      "fontSize": "22px",
                    }
                  }
                />
              </div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "padding": "20px 20px 20px 50px ",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "color": "Red",
                      "fontSize": "22px",
                      "paddingRight": "10px",
                    }
                  }
                >
                  Address:
                </div>
                <div
                  style={
                    Object {
                      "fontSize": "22px",
                    }
                  }
                />
              </div>
              <div
                style={
                  Object {
                    "display": "flex",
                    "padding": "20px 20px 20px 50px ",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "color": "Red",
                      "fontSize": "22px",
                      "paddingRight": "10px",
                    }
                  }
                >
                  PhoneNo:
                </div>
                <div
                  style={
                    Object {
                      "fontSize": "22px",
                    }
                  }
                />
              </div>
            </div>
          </CardPrint>
        </div>
        <div>
           
          <div
            id="testId"
          >
            0
          </div>
          <div
            id="testOnClick"
            onClick={[Function]}
          >
            Total Count
          </div>
        </div>
        <Component2
          action={[Function]}
          count={0}
        >
          <div>
            <button
              id="button"
              onClick={[Function]}
            >
              Count Total
            </button>
            <div>
              0
            </div>
            <Component1>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <div>
                  <div>
                    <div>
                      Yudhajit Adhikary
                    </div>
                    <div>
                      Koyena Adhikary
                    </div>
                  </div>
                </div>
                <div>
                  <div>
                    <div>
                      Blog
                    </div>
                    <div>
                      Blog1
                    </div>
                  </div>
                </div>
              </div>
            </Component1>
          </div>
        </Component2>
        <div>
          <Formik
            initialValues={
              Object {
                "address": "",
                "confirmPassword": "",
                "email": "",
                "password": "",
                "phoneno": "",
                "pincode": "",
              }
            }
            onSubmit={[Function]}
            render={[Function]}
            validationSchema={
              ObjectSchema {
                "_blacklist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "_conditions": Array [],
                "_defaultDefault": [Function],
                "_deps": Array [],
                "_excludedEdges": Array [],
                "_exclusive": Object {},
                "_mutate": undefined,
                "_nodes": Array [
                  "address",
                  "phoneno",
                  "pincode",
                  "confirmPassword",
                  "password",
                  "email",
                ],
                "_options": Object {
                  "abortEarly": true,
                  "recursive": true,
                },
                "_type": "object",
                "_typeError": [Function],
                "_whitelist": RefSet {
                  "list": Set {},
                  "refs": Map {},
                },
                "fields": Object {
                  "address": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": true,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                  "confirmPassword": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "required": true,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {
                        null,
                      },
                      "refs": Map {
                        "password" => Reference {
                          "getter": [Function],
                          "isContext": false,
                          "isSibling": true,
                          "isValue": false,
                          "key": "password",
                          "map": undefined,
                          "path": "password",
                        },
                      },
                    },
                    "_whitelistError": [Function],
                    "tests": Array [
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                  "email": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "email": false,
                      "required": true,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                  "password": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "matches": false,
                      "min": true,
                      "required": true,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                  "phoneno": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "matches": false,
                      "required": true,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                  "pincode": StringSchema {
                    "_blacklist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "_conditions": Array [],
                    "_deps": Array [],
                    "_exclusive": Object {
                      "matches": false,
                      "required": true,
                    },
                    "_mutate": undefined,
                    "_options": Object {
                      "abortEarly": true,
                      "recursive": true,
                    },
                    "_type": "string",
                    "_typeError": [Function],
                    "_whitelist": RefSet {
                      "list": Set {},
                      "refs": Map {},
                    },
                    "tests": Array [
                      [Function],
                      [Function],
                    ],
                    "transforms": Array [
                      [Function],
                    ],
                    "type": "string",
                  },
                },
                "tests": Array [],
                "transforms": Array [
                  [Function],
                ],
                "type": "object",
              }
            }
          >
            <Form>
              <form
                action="#"
                onReset={[Function]}
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="email"
                  >
                    Email
                  </label>
                  <Field
                    className="form-control"
                    name="email"
                    type="text"
                  >
                    <input
                      className="form-control"
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </Field>
                  <FormikConnect(ErrorMessageImpl)
                    className="invalid-feedback"
                    component="div"
                    name="email"
                  >
                    <ErrorMessageImpl
                      className="invalid-feedback"
                      component="div"
                      formik={
                        Object {
                          "dirty": false,
                          "errors": Object {},
                          "getFieldHelpers": [Function],
                          "getFieldMeta": [Function],
                          "getFieldProps": [Function],
                          "handleBlur": [Function],
                          "handleChange": [Function],
                          "handleReset": [Function],
                          "handleSubmit": [Function],
                          "initialErrors": Object {},
                          "initialStatus": undefined,
                          "initialTouched": Object {},
                          "initialValues": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                          "isSubmitting": false,
                          "isValid": true,
                          "isValidating": false,
                          "registerField": [Function],
                          "resetForm": [Function],
                          "setErrors": [Function],
                          "setFieldError": [Function],
                          "setFieldTouched": [Function],
                          "setFieldValue": [Function],
                          "setFormikState": [Function],
                          "setStatus": [Function],
                          "setSubmitting": [Function],
                          "setTouched": [Function],
                          "setValues": [Function],
                          "status": undefined,
                          "submitCount": 0,
                          "submitForm": [Function],
                          "touched": Object {},
                          "unregisterField": [Function],
                          "validateField": [Function],
                          "validateForm": [Function],
                          "validateOnBlur": true,
                          "validateOnChange": true,
                          "validateOnMount": false,
                          "values": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                        }
                      }
                      name="email"
                    />
                  </FormikConnect(ErrorMessageImpl)>
                </div>
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="password"
                  >
                    Password
                  </label>
                  <Field
                    className="form-control"
                    name="password"
                    type="password"
                  >
                    <input
                      className="form-control"
                      name="password"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="password"
                      value=""
                    />
                  </Field>
                  <FormikConnect(ErrorMessageImpl)
                    className="invalid-feedback"
                    component="div"
                    name="password"
                  >
                    <ErrorMessageImpl
                      className="invalid-feedback"
                      component="div"
                      formik={
                        Object {
                          "dirty": false,
                          "errors": Object {},
                          "getFieldHelpers": [Function],
                          "getFieldMeta": [Function],
                          "getFieldProps": [Function],
                          "handleBlur": [Function],
                          "handleChange": [Function],
                          "handleReset": [Function],
                          "handleSubmit": [Function],
                          "initialErrors": Object {},
                          "initialStatus": undefined,
                          "initialTouched": Object {},
                          "initialValues": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                          "isSubmitting": false,
                          "isValid": true,
                          "isValidating": false,
                          "registerField": [Function],
                          "resetForm": [Function],
                          "setErrors": [Function],
                          "setFieldError": [Function],
                          "setFieldTouched": [Function],
                          "setFieldValue": [Function],
                          "setFormikState": [Function],
                          "setStatus": [Function],
                          "setSubmitting": [Function],
                          "setTouched": [Function],
                          "setValues": [Function],
                          "status": undefined,
                          "submitCount": 0,
                          "submitForm": [Function],
                          "touched": Object {},
                          "unregisterField": [Function],
                          "validateField": [Function],
                          "validateForm": [Function],
                          "validateOnBlur": true,
                          "validateOnChange": true,
                          "validateOnMount": false,
                          "values": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                        }
                      }
                      name="password"
                    />
                  </FormikConnect(ErrorMessageImpl)>
                </div>
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="confirmPassword"
                  >
                    Confirm Password
                  </label>
                  <Field
                    className="form-control"
                    name="confirmPassword"
                    type="password"
                  >
                    <input
                      className="form-control"
                      name="confirmPassword"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="password"
                      value=""
                    />
                  </Field>
                  <FormikConnect(ErrorMessageImpl)
                    className="invalid-feedback"
                    component="div"
                    name="confirmPassword"
                  >
                    <ErrorMessageImpl
                      className="invalid-feedback"
                      component="div"
                      formik={
                        Object {
                          "dirty": false,
                          "errors": Object {},
                          "getFieldHelpers": [Function],
                          "getFieldMeta": [Function],
                          "getFieldProps": [Function],
                          "handleBlur": [Function],
                          "handleChange": [Function],
                          "handleReset": [Function],
                          "handleSubmit": [Function],
                          "initialErrors": Object {},
                          "initialStatus": undefined,
                          "initialTouched": Object {},
                          "initialValues": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                          "isSubmitting": false,
                          "isValid": true,
                          "isValidating": false,
                          "registerField": [Function],
                          "resetForm": [Function],
                          "setErrors": [Function],
                          "setFieldError": [Function],
                          "setFieldTouched": [Function],
                          "setFieldValue": [Function],
                          "setFormikState": [Function],
                          "setStatus": [Function],
                          "setSubmitting": [Function],
                          "setTouched": [Function],
                          "setValues": [Function],
                          "status": undefined,
                          "submitCount": 0,
                          "submitForm": [Function],
                          "touched": Object {},
                          "unregisterField": [Function],
                          "validateField": [Function],
                          "validateForm": [Function],
                          "validateOnBlur": true,
                          "validateOnChange": true,
                          "validateOnMount": false,
                          "values": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                        }
                      }
                      name="confirmPassword"
                    />
                  </FormikConnect(ErrorMessageImpl)>
                </div>
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="address"
                  >
                    Address
                  </label>
                  <Field
                    className="form-control"
                    name="address"
                    type="address"
                  >
                    <input
                      className="form-control"
                      name="address"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="address"
                      value=""
                    />
                  </Field>
                  <FormikConnect(ErrorMessageImpl)
                    className="invalid-feedback"
                    component="div"
                    name="address"
                  >
                    <ErrorMessageImpl
                      className="invalid-feedback"
                      component="div"
                      formik={
                        Object {
                          "dirty": false,
                          "errors": Object {},
                          "getFieldHelpers": [Function],
                          "getFieldMeta": [Function],
                          "getFieldProps": [Function],
                          "handleBlur": [Function],
                          "handleChange": [Function],
                          "handleReset": [Function],
                          "handleSubmit": [Function],
                          "initialErrors": Object {},
                          "initialStatus": undefined,
                          "initialTouched": Object {},
                          "initialValues": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                          "isSubmitting": false,
                          "isValid": true,
                          "isValidating": false,
                          "registerField": [Function],
                          "resetForm": [Function],
                          "setErrors": [Function],
                          "setFieldError": [Function],
                          "setFieldTouched": [Function],
                          "setFieldValue": [Function],
                          "setFormikState": [Function],
                          "setStatus": [Function],
                          "setSubmitting": [Function],
                          "setTouched": [Function],
                          "setValues": [Function],
                          "status": undefined,
                          "submitCount": 0,
                          "submitForm": [Function],
                          "touched": Object {},
                          "unregisterField": [Function],
                          "validateField": [Function],
                          "validateForm": [Function],
                          "validateOnBlur": true,
                          "validateOnChange": true,
                          "validateOnMount": false,
                          "values": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                        }
                      }
                      name="address"
                    />
                  </FormikConnect(ErrorMessageImpl)>
                </div>
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="pincode"
                  >
                    Pincode
                  </label>
                  <Field
                    className="form-control"
                    name="pincode"
                    type="pincode"
                  >
                    <input
                      className="form-control"
                      name="pincode"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="pincode"
                      value=""
                    />
                  </Field>
                  <FormikConnect(ErrorMessageImpl)
                    className="invalid-feedback"
                    component="div"
                    name="pincode"
                  >
                    <ErrorMessageImpl
                      className="invalid-feedback"
                      component="div"
                      formik={
                        Object {
                          "dirty": false,
                          "errors": Object {},
                          "getFieldHelpers": [Function],
                          "getFieldMeta": [Function],
                          "getFieldProps": [Function],
                          "handleBlur": [Function],
                          "handleChange": [Function],
                          "handleReset": [Function],
                          "handleSubmit": [Function],
                          "initialErrors": Object {},
                          "initialStatus": undefined,
                          "initialTouched": Object {},
                          "initialValues": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                          "isSubmitting": false,
                          "isValid": true,
                          "isValidating": false,
                          "registerField": [Function],
                          "resetForm": [Function],
                          "setErrors": [Function],
                          "setFieldError": [Function],
                          "setFieldTouched": [Function],
                          "setFieldValue": [Function],
                          "setFormikState": [Function],
                          "setStatus": [Function],
                          "setSubmitting": [Function],
                          "setTouched": [Function],
                          "setValues": [Function],
                          "status": undefined,
                          "submitCount": 0,
                          "submitForm": [Function],
                          "touched": Object {},
                          "unregisterField": [Function],
                          "validateField": [Function],
                          "validateForm": [Function],
                          "validateOnBlur": true,
                          "validateOnChange": true,
                          "validateOnMount": false,
                          "values": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                        }
                      }
                      name="pincode"
                    />
                  </FormikConnect(ErrorMessageImpl)>
                </div>
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="phoneno"
                  >
                    Contact Number
                  </label>
                  <Field
                    className="form-control"
                    name="phoneno"
                    type="phoneno"
                  >
                    <input
                      className="form-control"
                      name="phoneno"
                      onBlur={[Function]}
                      onChange={[Function]}
                      type="phoneno"
                      value=""
                    />
                  </Field>
                  <FormikConnect(ErrorMessageImpl)
                    className="invalid-feedback"
                    component="div"
                    name="phoneno"
                  >
                    <ErrorMessageImpl
                      className="invalid-feedback"
                      component="div"
                      formik={
                        Object {
                          "dirty": false,
                          "errors": Object {},
                          "getFieldHelpers": [Function],
                          "getFieldMeta": [Function],
                          "getFieldProps": [Function],
                          "handleBlur": [Function],
                          "handleChange": [Function],
                          "handleReset": [Function],
                          "handleSubmit": [Function],
                          "initialErrors": Object {},
                          "initialStatus": undefined,
                          "initialTouched": Object {},
                          "initialValues": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                          "isSubmitting": false,
                          "isValid": true,
                          "isValidating": false,
                          "registerField": [Function],
                          "resetForm": [Function],
                          "setErrors": [Function],
                          "setFieldError": [Function],
                          "setFieldTouched": [Function],
                          "setFieldValue": [Function],
                          "setFormikState": [Function],
                          "setStatus": [Function],
                          "setSubmitting": [Function],
                          "setTouched": [Function],
                          "setValues": [Function],
                          "status": undefined,
                          "submitCount": 0,
                          "submitForm": [Function],
                          "touched": Object {},
                          "unregisterField": [Function],
                          "validateField": [Function],
                          "validateForm": [Function],
                          "validateOnBlur": true,
                          "validateOnChange": true,
                          "validateOnMount": false,
                          "values": Object {
                            "address": "",
                            "confirmPassword": "",
                            "email": "",
                            "password": "",
                            "phoneno": "",
                            "pincode": "",
                          },
                        }
                      }
                      name="phoneno"
                    />
                  </FormikConnect(ErrorMessageImpl)>
                </div>
                <div
                  className="form-group"
                >
                  <button
                    className="btn btn-primary mr-2"
                    style={
                      Object {
                        "marginLeft": "50%",
                      }
                    }
                    type="submit"
                  >
                    Register
                  </button>
                </div>
              </form>
            </Form>
          </Formik>
        </div>
      </div>
    </Card>
  </Connect(Card)>
</Provider>
`;
